commit 2b48cc75b21431037d6f902b9d583b1aff198490
Author: Avi Kivity <avi@redhat.com>
Date:   Sat Nov 29 20:36:13 2008 +0200

    KVM: x86 emulator: fix popf emulation
    
    Set operand type and size to get correct writeback behavior.
    
    Signed-off-by: Avi Kivity <avi@redhat.com>

Adjusted to apply to Debian's 2.6.26 by dann frazier <dannf@debian.org>

diff -urpN linux-source-2.6.26.orig/arch/x86/kvm/x86_emulate.c linux-source-2.6.26/arch/x86/kvm/x86_emulate.c
--- linux-source-2.6.26.orig/arch/x86/kvm/x86_emulate.c	2010-02-04 22:07:32.000000000 -0700
+++ linux-source-2.6.26/arch/x86/kvm/x86_emulate.c	2010-02-04 22:17:43.000000000 -0700
@@ -1530,7 +1530,9 @@ special_insn:
 		emulate_push(ctxt);
 		break;
 	case 0x9d: /* popf */
+		c->dst.type = OP_REG;
 		c->dst.ptr = (unsigned long *) &ctxt->eflags;
+		c->dst.bytes = c->op_bytes;
 		goto pop_instruction;
 	case 0xa0 ... 0xa1:	/* mov */
 		c->dst.ptr = (unsigned long *)&c->regs[VCPU_REGS_RAX];
